app/controllers/like.js

```diff
@@ -0,0 +1,86 @@
+// Importation du model sauce
+const Sauce = require("../models/sauce");
+
+//...Liker et disliker une sauce
+exports.likeAndDislikeSauce = (req, res, next) => {
+  console.log(req.body.like);
+
+  // Aller chercher l'objet dans la base de données (méthode findOne)
+  //(req.params pour récupérer l'id de la sauce dans la requête et attribuer cette valeur à la key _id)
+  Sauce.findOne({ _id: req.params.id })
+    .then((sauce) => {
+      console.log("contenu result promise : sauce");
+      console.log(sauce);
+
+      // Switch case
+      switch (req.body.like) {
+        // AJOUTER UN LIKE (like +1)
+        case 1:
+          if (
+            !sauce.usersLiked.includes(req.body.userId) &&
+            req.body.like === 1 && !sauce.usersDisliked.includes(req.body.userId)
+          ) {
+            // Mise à jour sauce dans la base de données
+            Sauce.updateOne(
+              // Chercher la sauce dans la base de données
+              { _id: req.params.id },
+              {
+                // Incrementation 1 dans le champ likes
+                $inc: { likes: 1 },
+                // Ajouter le userId dans le tableau des usersLiked
+                $push: { usersLiked: req.body.userId },
+              }
+            )
+              .then(() => res.status(201).json({ message: "Sauce like +1" }))
+              .catch((error) => res.status(400).json({ error }));
+          }
+          break;
+
+        // AJOUTER UN DISLIKE (dislike +1)
+        case -1:
+          if (
+            !sauce.usersDisliked.includes(req.body.userId) &&
+            req.body.like === -1 && !sauce.usersLiked.includes(req.body.userId)
```

Comment:
req.body.like is -1 because it's your case, so not necessary. Then you muste have another if with sauce.usersLiked.includes(req.body.userId) then remove userId from usersLiked array